Uyuni salt config modules (#523)
uyuni-project/uyuni-docs: modules/salt/pages/config-modules.adoc

[[config-modules]]
= Configuration Modules

[IMPORTANT]
====
This feature is a technology preview.
====

> You can use Salt execution and state modules to configure your {productname} Server.

Author: Hmm, not sure if this is the best wording.

So these modules are called "Uyuni configuration modules" and they can target both Uyuni and SUSE Manager Servers.

As it is, this sentence seems to refer to generic Salt execution and state modules: of course one could use them to do whatever, including Server configuration, but here we are presenting some specific new modules that are tailored for that scope.

I'd still call them Uyuni configuration modules (just as the package name) and clarify they also apply to SUSE Manager contexts.

Reviewer: OK, that's what had me confused! I've updated the wording now, so hopefully it's clearer.

These configuration modules can be used directly or using SLS files.
They are are especially useful if you have multiple {productname} Servers, for example in Hub installations, but they are also useful for smaller installations.

For more information about using Hub, see xref:large-deployments:multi-server.adoc[].

You can use configuration modules to configure:

* Organizations
* Users
* User permissions
* System groups

> More configuration modules are planned to be released in the future.

Author: Technically speaking there is no plan set in stone yet, although I can see them coming. Not even sure if it's worth it to mention here...

Reviewer: Removed.

> * For more information about Salt execution modules and how to use them, see https://docs.saltstack.com/en/latest/topics/tutorials/modules.html.

Reviewer 2: ```suggestion
* For more information about Salt execution modules, see https://docs.saltstack.com/en/latest/topics/tutorials/modules.html.
```

* For more information about Salt state modules, see https://docs.saltstack.com/en/latest/topics/tutorials/starting_states.html.

== Install Configuration Modules

> The Salt modules are available in the [path]``uyuni-config-modules`` package.

Author: Again this sentence does not sound correct: [many](https://docs.saltstack.com/en/latest/ref/states/all/index.html) [many](https://docs.saltstack.com/en/latest/ref/modules/all/index.html) Salt modules exist and most of them are not part of that package. The sentence is supposed to refer only to the Uyuni configuration states.

Reviewer 2: I think we have a `[package]` tag.

[[config-modules]]
= Configuration Modules

[IMPORTANT]
====
This feature is a technology preview.
====

> Salt execution and state modules are used to configure your installation.

Author: ```suggestion
Execution and state modules are the main concepts in Salt to define, apply and orchestrate configuration of your devices.
```

Reviewer: Slightly edited this to make it clearer, what do you think?

Author: :+1:

{productname} provides a set of modules called {uyuni} configuration modules, that you can use to configure both {susemgr} and {uyuni} Servers.

You can use the {uyuni} configuration modules directly or using SLS files.
They are are especially useful if you have multiple {productname} Servers, for example in Hub installations, but they are also useful for smaller installations.

For more information about using Hub, see xref:large-deployments:multi-server.adoc[].

> You can use configuration modules to configure:

Author: ```suggestion
You can use {uyuni} configuration modules to configure:
```
